react/week3/ToDoList.js

```diff
@@ -0,0 +1,165 @@
+import React, { useEffect, useState } from "react";
+import PropTypes from "prop-types";
+// styling the border
+const TodoComponent = (props) => {
+  return (
+    <div
+      style={{
+        width: "700px",
+        marginLeft: "300px",
+        fontSize:"20px",
+        fontFamily:"cursive",
+      }}
+    >
+      {props.children}
+    </div>
+  );
+};
+const API_URL =
+  "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+function TodoList({ id, description, deadline, removeTodo }) {
+  const [isDone, setIsDone] = useState(false);
+  const [edit, setEdit] = useState(false);
+  const [newDescription, setNewDescription] = useState(description);
+  const updateTodo = () => {
+    setEdit(!edit);
+  };
+  return (
+    <>
+      <TodoComponent>
+        <div
+          className="list-of-todos"
+          style={
+            isDone ? { textDecoration: "line-through" } : { textDecoration: "" }
+          }
+        >
+          {edit ? (
+            <input
+              type="text"
+              value={newDescription}
+              onChange={(event) => {
+                const value = event.target.value;
+                setNewDescription(value);
+              }}
+            />
+          ) : (
+            newDescription
+          )}{" "}
+          | {deadline}
+          <div className="buttons">
+            <input type="checkbox" onClick={() => setIsDone(!isDone)} />
+            <button className="delet-button" onClick={() => removeTodo(id)}>
+              Delete
+            </button>
+            <button className="edit-button" onClick={updateTodo}>
+              {edit ? "Update" : "Edit"}
+            </button>
+          </div>
+        </div>
+      </TodoComponent>
+    </>
+  );
+}
+//component is for input elements
+function RenderTodos() {
+  const [description, setDescription] = useState("");
+  const [deadline, setDeadline] = useState("");
+  const [todos, setTodos] = useState([]);
+
+  const minimumDate = () => {
+    let today = new Date();
+    let dd = today.getDate();
+    let mm = today.getMonth() + 1; 
+    let yyyy = today.getFullYear();
+    if (dd < 10) {
+      dd = "0" + dd;
+    }
+    if (mm < 10) {
+      mm = "0" + mm;
+    }
+    today = yyyy + "-" + mm + "-" + dd;
+    return today;
+  };
```

Comment:
There are lot of methods available on the Date object that can come handy in cases like this. 
If you want to make this shorter, you can also just write it like this:
`new Date().toLocaleDateString("en-GB").replaceAll("/", "-")`
It will give you the same result.